pkg/store/bucket.go

```diff
@@ -1394,6 +1394,11 @@ func (r *bucketIndexReader) loadSeries(ctx context.Context, ids []uint64, start,
 	r.stats.seriesFetchedSizeSum += int(end - start)
 
 	for _, id := range ids {
+
+		if id-start > uint64(len(b)) {
```

Comment:
Hm are we should this is the correct place to validate against malformed index?

If we are asking to fetch `start -> end-start` from `r.block.readIndexRange(ctx, int64(start), int64(end-start))` I think this method should FAIL if there is no such range in the object storage. So I believe we should fix that method instead of every small usages of those in thousands of places like this. (: 

Also what is the root cause again? Because you have some block that has malformed index right? Wonder in what way malformed.

Reply:
This is also what I was thinking. This is a good stop to the problem but not a good fix. It might not even be the correct fix at all.

The root cause looks to be that some blocks have a malformed index. I should have the results of a `thanos bucket verify` at some point today or tomorrow.